plugins/catalog-import/src/util/useGithubRepos.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as YAML from 'yaml';
+import { useApi, githubAuthApiRef } from '@backstage/core';
+import { catalogImportApiRef } from '../api/CatalogImportApi';
+import { ConfigSpec } from '../components/ImportComponentPage';
+
+export function useGithubRepos() {
+  const api = useApi(catalogImportApiRef);
+  const auth = useApi(githubAuthApiRef);
+
+  const submitPRToRepo = async (selectedRepo: ConfigSpec) => {
+    const token = await auth.getAccessToken(['repo']);
```

Comment:
I've been thinking about this. What are peoples thoughts about using either backend as a proxy with the GITHUB_TOKEN for scenarios where readonly GH API access is only required? Having to setup GH OAuth and also introducing multiple login providers (in some cases) seems to create a higher barrier of entry.

Reply:
We want the frontend plugin to work in behalf of the user's github account to create pull requests as them.

Reply:
That's a fair point for sure, but also creates another barrier. Using an OAuth token in this manner likely requires a GH org admin to be involved. So I could see this being more contentious and or time consuming to setup in some organizations. I'm not necessarily suggesting to change it here, but pointing out some downsides to the GH OAuth approach. I think approaching the API aspect of this plugin from a perspective where it "could" use a backend implemented API in the future would be ideal. That said that can be a future consideration as well.

Reply:
We have this situation at Spotify. The login provider is a specialised one that decodes a Google IAP token and looks up user info in LDAP, so there's a Google login screen to get past for visiting the page, but it happens extremely rarely since the session is picked up and renewed silently with a very long expiry. Then when you want to perform some action that requires GitHub Enterprise access (like this PR), the FIRST time you will be asked for consent and then pretty much never again since we make use of refresh tokens. So the cost for the end user is one login screen and one consent popup extremely seldom. The cost for the integrator is indeed to need an OAuth application to be added to their GitHub (if that's what they use), but that's a one time cost. The token that the proxy or backend would use, also needs the GitHub owner's involvement to get, right?

Reply:
@freben - I'm no expert, but based on my experimentation, the OAuth token and standard GH user tokens are not equivalent. We've been able to use a standard (non GH org admin) user token for the backend integrations without any problems. However, the only way to get plugins that use GH auth to work on private repos is to have a GH org admin create the OAuth token. All that said, I'm not hung up on this point, just having an open conversation 😄

Reply:
@andrewthauer Tricky bit would be to seamlessly transition between the two. Integrations work is heading in that direct along with `UrlReader`s etc. The possibility of a `RepoApi` is something we've surfaced a couple of times, which could make it a bit easier to work with repos without having to care exactly how authentication happens or where the repos reside. There's plenty more plumbing to get in place before we'd be able to reach something like that though, as far as I can tell.

Reply:
I agree. It's not a problem for us at the moment. Although we have a different primary auth provider then GH, so we need to support 2 auth providers to gain some features via the GH OAuth. It's fine, just not ideal. It's really nice that backstage prompts when GH auth is required though 😄.